cura/CuraPackageManager.py

```diff
@@ -99,64 +113,66 @@ def getInstalledPackageInfo(self, package_id: str) -> Optional[dict]:
             return package_info
 
         if package_id in self._installed_package_dict:
-            package_info = self._installed_package_dict.get(package_id)
+            package_info = self._installed_package_dict[package_id]["package_info"]
             return package_info
 
-        for section, packages in self.getAllInstalledPackagesInfo().items():
-            for package in packages:
-                if package["package_id"] == package_id:
-                    return package
+        if package_id in self._bundled_package_dict:
+            package_info = self._bundled_package_dict[package_id]["package_info"]
+            return package_info
 
         return None
 
     def getAllInstalledPackagesInfo(self) -> dict:
-        installed_package_id_set = set(self._installed_package_dict.keys()) | set(self._to_install_package_dict.keys())
-        installed_package_id_set = installed_package_id_set.difference(self._to_remove_package_set)
 
-        managed_package_id_set = installed_package_id_set | self._to_remove_package_set
+        # Add bundled, installed, and to-install packages to the set of installed package IDs
+        all_installed_ids = set()
 
-        # TODO: For absolutely no reason, this function seems to run in a loop
-        # even though no loop is ever called with it.
+        if self._bundled_package_dict.keys():
+            all_installed_ids = all_installed_ids.union(set(self._bundled_package_dict.keys()))
+        if self._installed_package_dict.keys():
+            all_installed_ids = all_installed_ids.union(set(self._installed_package_dict.keys()))
+        if self._to_install_package_dict.keys():
+            all_installed_ids = all_installed_ids.union(set(self._to_install_package_dict.keys()))
+        all_installed_ids = all_installed_ids.difference(self._to_remove_package_set)
 
         # map of <package_type> -> <package_id> -> <package_info>
         installed_packages_dict = {}
-        for package_id in installed_package_id_set:
+        for package_id in all_installed_ids:
+
+            # Skip required plugins as they should not be tampered with
             if package_id in Application.getInstance().getRequiredPlugins():
```

Comment:
It's kind of strange to not list an item because it's required... but the getAllInstalledPackagesInfo function is currently only used in (un)installing. oh well.

Reply:
This was a decision made a while back that since required plugins should never be "managed" in that you shouldn't be able to disable or remove them, there was no reason displaying them.